AsyncDisplayKit/ASMapNode.mm

```diff
@@ -185,9 +191,23 @@ - (void)removeLiveMap
     _mapView = nil;
     _mapNode = nil;
   }
-  self.image = nil;
 }
 
+- (void)setAnnotations:(NSArray *)annotations
+{
+    ASDN::MutexLocker l(_propertyLock);
+    _annotations = [annotations copy];
+    if (annotations.count != _annotations.count) {
```

Comment:
@appleguy I've rewritten this so that it works correctly, both when we have a liveMap and also when only the snapshot is being displayed.

In terms of getting rid of the boolean, I've got rid of _liveMap and now use self.isLiveMap exclusively, rather than mixing and matching _liveMap && !_mapView. It's a lot better that way, thanks for that!